syntax = "proto3";

package proto;

import "compiler/grpc/proto/common.proto";
import "compiler/grpc/proto/competition.proto";
import "compiler/grpc/proto/requests.proto";
import "compiler/grpc/proto/round.proto";
import "compiler/grpc/proto/season.proto";
import "compiler/grpc/proto/team.proto";
import "compiler/grpc/proto/venue.proto";

service FixtureService {
    rpc ListSeasonFixtures(SeasonFixtureRequest) returns (stream Fixture) {}
    rpc FixtureByID(FixtureRequest) returns (Fixture) {}
}

message Fixture {
    int64 id = 1;
    Competition competition = 2;
    Season season = 3;
    Team home_team = 4;
    Team away_team = 5;
    Round round = 6;
    Venue venue = 7;
    Date date_time = 8;
}
